libs/designsystem/card/src/card.component.spec.ts

```diff
@@ -74,7 +74,7 @@ describe('CardComponent', () => {
       const ionBadgeElement = spectator.query<HTMLElement>('ion-badge');
       expect(ionBadgeElement).toBeTruthy();
       await TestHelper.whenReady(ionBadgeElement);
-      expect(ionBadgeElement).toHaveClass('hydrated', ionicGlobalMode);
+      expect(ionBadgeElement).toHaveClass('ios', ionicGlobalMode);
```

Comment:
Efter skiftet til standalone begyndte denne test at fejle på at klassen var ios i stedet for hydrated.
Jeg er usikker på om det er ønsket opførsel eller ej.

Reply:
I think we should still wrap it in a promise, so that the promise returned from the setPart method resolves after all the work is done.

We could create a util function that wraps the ionic function and returns a promise:
```ts
import { componentOnReady as ionicComponentOnReady } from '@ionic/core';

export async function componentOnReady(element: Element): Promise<Element> {
  return await new Promise<Element>((resolve) => {
    ionicComponentOnReady(element, resolve);
  });
}
```

And then use it like this:
```ts
  public async setPart(partName: string, ionicElementRef: ElementRef, selector: string) {
    // Ensure custom element has been defined and rendered
    await customElements.whenDefined(ionicElementRef.nativeElement.localName);
    const customElement = await componentOnReady(ionicElementRef.nativeElement);

    const partElement: HTMLElement = customElement.shadowRoot.querySelector(selector);
    if (!partElement) return;

    this.renderer.setAttribute(partElement, 'part', partName);
  }
```